### PR TITLE
Geometry update for 2022

### PR DESCRIPTION
**Pull request by klendathu2k**
- Implement first cut of the y2022 geometry @ timestamp 20211010.000000
-- sTGC and forward silicon detectors are still in development / expecting to fold in updates as they become available
-- overlaps are a concern for the initial geometries

- Remove deprecated dev2021 geometry model
- Advance dev2022 timestamp by 100 years to not interfere

### CODE REVIEW

**Comment by klendathu2k** [comment]
S&C Management Meeting discussion, it is time to create the y2022 first cut geometry.

**Comment by genevb** [comment]
I see that we did the 15th of the month in the past for the geometry timestamp (the 10th of the month was for development?). So perhaps we should set it to 20211015?

**Comment by fvidebaek** [comment]
Jason  if yu removed dev2021 which is fine, we need to discuss how to do simplified vs. ful geometry. Can be talk on  Friday (I am busy tih DNP tomorrow)

F

**Comment by klendathu2k** [comment]
@genevb I suppose consistency would be nice here.  I'll update to the 15th.

@fvidebaek Good point. The reason I removed dev2021 was to avoid a conflict with timestamps... but I can do the same trick as with dev2022.

Codes will be pushed later this evening.

File: StarVMC/Geometry/StarGeo.xml [hunk 1]
```diff
@@ -61,15 +60,15 @@
...
     <Construct sys="magp" config="MAGPon"  />
     <Construct sys="mutd" config="MUTD16"  />
-    <Construct sys="fstm" config="FSTMv1" />
-    <Construct sys="stgm" config="STGCv1" />
-    <!-- Construct sys="etof" config="ETOFv14f"  -->
     <Construct sys="wcal" config="WCALv1" simu="1" />
     <Construct sys="hcal" config="HCALv1" simu="1" />
-    <Construct sys="plat" config="PLATon" simu="2" />
-    <Construct sys="pres" config="PRESof" simu="2" /> 
+    <!-- Note:  These versions of the forward trackers have significant overlaps which need to be resolved -->
```
Comment (fvidebaek) [inline StarVMC/Geometry/StarGeo.xml]: Not quite sure I understand the overlap comment for FSTMv1. I agree that the STGCv1 is not correct. It is being worked on.

**Comment by klendathu2k** [comment]
Mainly a warning / reminder to myself that these detector models need to 
be updated.
Agreed, sTGC is the main concern.

File: StRoot/StChain/GeometryDbAliases.h [hunk 2]
```diff
@@ -155,9 +155,11 @@ static const DbAlias_t fDbAlias[] = {// geometry  Comment            old
 
   {"y2021",       20201215,     0, "y2021",    "y2021 development geometry, AgML,xgeometry"},   
   {"y2021a",      20201215,     1, "y2021a",   "y2021 first production geometry, AgML,xgeometry"},   
-  
-  {"dev2021",     20221210,     0, "dev2021",  "-deprecated- geometry for 2021+ forward program,AgML,xgeometry"},
-  {"dev2022",     20221210,     1, "dev2022",  "development geometry for 2022+ forward program,AgML,xgeometry"},
+
+  {"y2022",       20211010,     0, "y2021",    "y2021 development geometry, AgML,xgeometry"},
```
Comment (fvidebaek) [inline StRoot/StChain/GeometryDbAliases.h]: line 159 - shouldn't the 2 last y2021 be 2022 ?

**Review by fvidebaek — COMMENTED** [review]
question on line 159 . in .age

File: StRoot/StChain/GeometryDbAliases.h (hunk 2 above)
Comment (klendathu2k) [reply StRoot/StChain/GeometryDbAliases.h]: Yes it should... also, missed an update on xgeometry which is now in place.

Comment (klendathu2k) [reply StRoot/StChain/GeometryDbAliases.h]: Clearly haven't had enough coffee today...

**Review by fvidebaek — APPROVED** [review]

**Review by genevb — APPROVED** [review]
I am not reviewing most of what is being changed here, but I will approve the addition of y2022 to GeometryDbAliases.h with the 20211015 date.

**Comment by klendathu2k** [comment]
@perevbnlgov I'd like to merge this, please review changes.

File: StRoot/StChain/GeometryDbAliases.h [hunk 3]
```diff
@@ -155,9 +155,11 @@ static const DbAlias_t fDbAlias[] = {// geometry  Comment            old
...
   {"y2021",       20201215,     0, "y2021",    "y2021 development geometry, AgML,xgeometry"},   
   {"y2021a",      20201215,     1, "y2021a",   "y2021 first production geometry, AgML,xgeometry"},   
-  
-  {"dev2021",     20221210,     0, "dev2021",  "-deprecated- geometry for 2021+ forward program,AgML,xgeometry"},
-  {"dev2022",     20221210,     1, "dev2022",  "development geometry for 2022+ forward program,AgML,xgeometry"},
+
+  {"y2022",       20211015,     0, "y2022",    "y2022 first cut geometry, AgML,xgeometry"},   
+
+  {"dev2021",     21201210,     1, "dev2021",  "-deprecated- geometry for 2021+ forward program,AgML,xgeometry"},  
+  {"dev2022",     21211210,     1, "dev2022",  "development geometry for 2022+ forward program,AgML,xgeometry"},
```
Comment (plexoos) [inline StRoot/StChain/GeometryDbAliases.h:162]: Are the first four digits in the second column supposed to represent the year? Is it intentional to have `2120` and `2121`?

**Comment by klendathu2k** [comment]
I have advanced the (soon-to-be-deprecated) "development" geometries by 
100 years so that the timestamps do not interfere
with the production geometries.

**Review by plexoos — APPROVED** [review]

**Comment by perevbnlgov** [comment]
No objections

**Pull request merged**